### PR TITLE
added forgot password feature

### LINKED ISSUE #22: Forgot password Feature
### Expected Behavior

After adding forgot password feature user is supposed to enter his/her's email id, then the email will be checked in the database and if it is there then an email will be sent to the user providing them with an OTP( random 4 digit number), and in parallel with the email, the user will be redirected to an another page where he will enter the otp and if the otp matches with the generated otp we will redirect them to a new page to change their password.

### Current Behavior

There is no way to change password if user forgets it.

I would like to add this feature to the project @vikasdo please assign me to do the same.

### PR DESCRIPTION
**Pull request by VishvamPorwal**
# Description

Added forgot password feature as described in the issue.
Closes #22 

## Type of change

Added forgot password feature.

# How Has This Been Tested?

Tried this feature with multiple emails.

### CODE REVIEW

**Review by Akshay6890 — CHANGES_REQUESTED** [review]

File: bookstore/client/views.py [hunk 1]
```diff
@@ -388,3 +389,60 @@ def discounts():
...
+            generated_otp = random.randint(1000, 9999)
+            session["generated_otp"] = generated_otp
+            try:
+                message = Message("Here is your Otp",
+                                  sender='bookly1120@gmail.com', recipients=[email])
+                message.body = f"Your otp is {generated_otp}"
+                mail.send(message)
+            except:
+                flash(f"Error sending email to {email}")
+            return redirect(url_for("otp"))
```
[the remark this answers is not in the document]
Comment (VishvamPorwal) [reply bookstore/client/views.py]: fixed it, sorry about that I guess I forgot about it.

File: bookstore/client/views.py [hunk 2]
```diff
@@ -388,3 +389,60 @@ def discounts():
...
     return redirect(url_for('personalized_offers'))
+
+
+@app.route("/forgotpwd", methods=["POST", "GET"])
+def forgotpwd():
+    if request.method == "GET":
+        return render_template("client/forgot_password.html")
+    else:
+        email = request.form.get("email")
+        session["cli_email"] = email
```
[the remark this answers is not in the document]
Comment (VishvamPorwal) [reply bookstore/client/views.py:400]: Pardon me, but I am not getting your point, how would user login if they dont remember there password.

**Comment by Akshay6890** [comment]
hey that session comment was a mistake, you did it right :)

File: bookstore/client/views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Akshay6890) [reply bookstore/client/views.py:400]: > Hey, ignore that session, it was my mistake :)

**Comment by VishvamPorwal** [comment]
I have made some commits to fix every issue.

**Comment by Akshay6890** [comment]
Ok, this looks great!

**Review by Akshay6890 — APPROVED** [review]
This is great!

**Comment by VishvamPorwal** [comment]
I did not receive any marks from this pr.

**Pull request merged**